torch/distributed/rendezvous.py

```diff
@@ -151,6 +181,10 @@ def _create_c10d_store(hostname, port, rank, world_size, timeout) -> Store:
     if not 0 <= port < 2**16:
         raise ValueError(f"port must have value from 0 to 65535 but was {port}.")
 
+    # if world_size is None then there is a non-fixed number of members
+    if not world_size:
+        world_size = -1
```

Comment:
(can be done in followup PR): If it is designated to use `optional<int>` and `None` to indicate dynamic group (as in TensorPipeAgent), let's make it consistent everywhere (i.e., modify TCPStore accordingly).

Reply:
Hold on, looks like TCPStore already accepts None?

https://github.com/pytorch/pytorch/blob/f0db315efeef27a2d8760cec4573bcbe40854b01/torch/csrc/distributed/c10d/TCPStore.cpp#L964

Reply:
It does but the pybind definition does not. I added an issue to track. https://github.com/pytorch/pytorch/issues/74488